### PR TITLE
CI: Add pre-commit CI

### PR DESCRIPTION
**Pull request by CSSFrancis**
#### CI: Add pre-commit CI

This adds lines to pre-commit for CI to run isort and black from a comment to a PR.

#### For reviewers

- [x] The PR title is short, concise, and will make sense 1 year later.
- [x] New functions are imported in corresponding `__init__.py`.
- [x] New features, API changes, and deprecations are mentioned in the unreleased
      section in `CHANGELOG.rst`.
- [x] Contributor(s) are listed correctly in `__credits__` in `orix/__init__.py` and in
      `.zenodo.json`.

### CODE REVIEW

**Comment by pc494** [comment]
pre-commit.ci autofix

**Comment by pc494** [comment]
at a guess this failure is due to the style changes between 2023 and 2024. @CSSFrancis could you run locally so this comes up green as I imagine we need this merged before we can autofix.

**Comment by CSSFrancis** [comment]
@pc494 Yep you are correct. The joys of running `black` :)

**Comment by CSSFrancis** [comment]
@pc494 This should be fixed now and we can auto fix now which is nice.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 repos:
   - repo: https://github.com/psf/black
-    rev: 23.1.0
+    rev: 24.3.0
```
Comment (pc494) [inline .pre-commit-config.yaml:3]: Good.

**Review by pc494 — APPROVED** [review]

**Comment by pc494** [comment]
I am going to merge this on the condition that @CSSFrancis remembers to add himself to the Zenodo when we get to #460. This PR closes #463.

**Pull request merged**